bin/cml-runner.js

```diff
@@ -379,11 +387,6 @@ const opts = yargs
     'cloud-ssh-private',
     'Custom private RSA SSH key. If not provided an automatically generated throwaway key will be used'
   )
-  .boolean('cloud-ssh-private-visible')
```

Comment:
@0x2b3bfa0 if we remove the functionality of showing the ssh we HAVE to remove the capability of creating the SSH automatically with all the friction that it generates. Makes no sense that we are able to create SSH keys and nt being able to give back to them.
I think that giving them the capability of showing it at their own risk is acceptable

Reply:
> _Let's do it. They can specify the SSH so it might be fine._

Awesome! In my opinion, specifying the SSH private key is, if not the best solution, at least a good short term compromise between printing keys to the logs and having users do everything manually.

> _The rationale of my original thought is because there are some users that might not know how to generate SSH keys (I found them), so I wanted to provide them it back._

If that's the case, let's document the process:

1. Generate<sup>2</sup> a new SSH RSA (PEM) private key:

   ```console
   $ ssh-keygen -t rsa -m pem -b 4096 -f key.pem
   ```

2. Pass the `--cloud-ssh-private` to `cml-runner` including the contents<sup>1</sup> of newly genereated private key file:

   ```console
   $ cml-runner --cloud ··· --cloud-ssh-private="$(cat key.pem)"
   ```

***

<sup>1</sup> Would be a bit more involved if we don't resort to the changes introduced with #602. Using `perl` or `sed` for this is not precisely an user-friendly solution.
<sup>2</sup> Hardcoding key bit lengths in documentation is not future-proof, but perhaps better than recommending the default 3072 bit length.

Reply:
🔔  @casperdcl, should we add this to the wiki under the `cml-runner` reference section?

Reply:
I would say no. What you are doing with that is making users generate the "useless" keys by themselves so the inbuilt feature is not needed anymore

Reply:
> It does make sense as long as we keep using them internally for log retrieval in some platforms as per iterative/terraform-provider-iterative#70. Anyhow.

We cant do this without breaking our users workflows. WE have to just keep the inbuilt SSH creation but not even letting know how to generate the SSH key by themselves, there are plenty of tutorials for that

Reply:
👌🏼

Reply:
> should we add this to the wiki

please do!

Reply:
[Done!](https://github.com/iterative/cml/wiki/ref:-cml-runner) It leaves room for improvement, but at least it won't get lost.

Reply:
I do not fully understand why we need that. Docs says:
```
Custom private RSA SSH key. If not provided an
                               automatically generated throwaway key will be
                               used
```
Additionally we are teaching them how to create a throwaway key? @casperdcl @0x2b3bfa0

Reply:
If users take no action, the Terraform provider will automatically generate a throwaway key for internal use. This key won't be available to users and will only be used to retrieve machine logs and determine the runner status.

If users need to access the instance on their own, they will need to know how to provide their own key. As you can guess, we are also teaching them how to generate a new one because using their default key from `.ssh/id_rsa`:

1.  Might not be an option, as it's usually in OpenSSH format, not in PEM format.
2. Would expose the user private key beyond what's acceptable in many scenarios.